staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -196,6 +196,13 @@ type ValidationRule struct {
 	// If unset, the message is "failed rule: {Rule}".
 	// e.g. "must be a URL with the host matching spec.host"
 	Message string
+	// MessageExpression represents a CEL expression that can be evaluated when validation fails, whereas Message
```

Comment:
I wouldn't bring up Message here since it is already covered in detail below.

Reply:
Gotcha, I removed that bit.